### PR TITLE
#147 implement use-simple-attribute

### LINKED ISSUE #147: new rule: enforce simple TSX attributes
new rule: enforce simple TSX attributes

Among the things that you should allow in tsx attributes: 
- variables
- properties
- method invocations
- binary expressions on constants ( className={ "container" + "whatever" }
- binary expressions with a single constant ( className={ "container" + myClassName }

Things to avoid: 
- ternary expressions
- complex binary expressions ( className =  { someName + '-' + someOtherName } )

### PR DESCRIPTION
**Pull request by noamyogev84**
Fixes #147 
* implement use-simple-attribute

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -185,6 +185,7 @@
     "react-a11y-lang": true,
     "react-a11y-meta": true,
     "react-a11y-no-onchange": true,
+    "use-simple-attribute": true,
```
Comment (JoshuaKGoldberg) [inline tslint.json]: Sorry to nitpick, but could you please keep these rules alphabetized? I find it convenient to be able to quickly scan through them.

File: src/useSimpleAttributeRule.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
+import * as ts from 'typescript';
+import * as Lint from 'tslint';
+
+import {ExtendedMetadata} from './utils/ExtendedMetadata';
+import {getJsxAttributesFromJsxElement} from './utils/JsxAttribute';
+
+export class Rule extends Lint.Rules.AbstractRule {
+
+    public static metadata: ExtendedMetadata = {
```
Comment (JoshuaKGoldberg) [inline src/useSimpleAttributeRule.ts]: Could you please also add a `rationale`? It's not clear from just the description _why_ we would want such a thing.

File: src/tests/useSimpleAttributeRuleTests.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
+import {Utils} from '../utils/Utils';
+import {TestHelper} from './TestHelper';
+/**
+ * Unit tests.
+ */
+describe('useSimpleAttributeRule', () : void => {
+    const ruleName : string = 'use-simple-attribute';
+    const binaryExpressionErrorMessage: string = 'Attribute containes a complex binary expression';
+    const trenaryExpressionErrorMessage: string = 'Attribute contains a trenary expression';
```
Comment (JoshuaKGoldberg) [inline src/tests/useSimpleAttributeRuleTests.ts]: > trenary

Typo: `ternary`

File: tslint.json (hunk 1 above)
Comment (noamyogev84) [reply tslint.json]: done.

File: src/tests/useSimpleAttributeRuleTests.ts (hunk 3 above)
Comment (noamyogev84) [reply src/tests/useSimpleAttributeRuleTests.ts]: done.

File: src/useSimpleAttributeRule.ts (hunk 2 above)
Comment (noamyogev84) [reply src/useSimpleAttributeRule.ts]: Well, i tried to look for additional info about that with no success. help will be appreciated.

Comment (JoshuaKGoldberg) [reply src/useSimpleAttributeRule.ts]: Sure! Something like...

> Simpler attributes in JSX and TSX files helps keep code clean and readable.
> Separate complex expressions into their own line and use clear variable names to make your code more understandable

...maybe?

Comment (noamyogev84) [reply src/useSimpleAttributeRule.ts]: @JoshuaKGoldberg done!
Thanks!

**Comment by noamyogev84** [comment]
Hi @JoshuaKGoldberg. anything else on my side?

**Comment by JoshuaKGoldberg** [comment]
Hey @noamyogev84 - the `master` branch has been updated since you last sent this PR and there are some merge conflicts. They look pretty small, so you should be able to resolve them with the GitHub 'Resolve conflicts' button without too much hassle. @ me if that's not the case and I can take a look. 😊

**Comment by noamyogev84** [comment]
@Hi @JoshuaKGoldberg .I've completed the merge. I also saw that you've removed the `PR: Waiting for Author` tag. Can I further assist here?

**Comment by JoshuaKGoldberg** [comment]
Thanks for the ping @noamyogev84! PR is ready to merge.

You had a couple files that had duplicated lines from the merge, so I went ahead and removed them for you. _Then_ I noticed there were a couple of small nitpicks I'd missed in the review _(rule should have a plural name per its description; update the description in README.md as well)_ and didn't want to be annoying and ask you to do them after the review, so I hope you don't mind me having submitted them myself 😊.

**Comment by noamyogev84** [comment]
@JoshuaKGoldberg Thanks. I appreciate your attitude and help!

**Pull request merged**